crates/nu-protocol/src/ast/cell_path.rs

```diff
@@ -173,22 +174,46 @@ impl CellPath {
             member.make_optional();
         }
     }
+
+    // Formats the cell-path as a column name, i.e. without quoting and optional markers ('?').
+    pub fn to_column_name(&self) -> String {
+        let mut s = String::new();
+
+        for member in &self.members {
+            match member {
+                PathMember::Int { val, .. } => {
+                    s += &val.to_string();
+                }
+                PathMember::String { val, .. } => {
+                    s += val;
```

Comment:
Here we still need quotes around number-like columns otherwise they will become indistinguishable from rows.

Reply:
Yeah, but is that a problem in this situation? Since all the column names get crammed into a string anyway, I think it's easier for users to not quote indexes, e.g. `get 'a."0".x'` vs `get a.0.x`

Reply:
`0`- `PathMember::Int` -> row
`"0"` - `PathMember::String` -> column

It would be impossible to determine the meaning of something like 

```nushell
let confusing = [{"0": bla, "1": blub}, {"0": foo, "1": bar}]
let cellp = $.0."1"

print $cellp
$confusing | select $cellp
```

Reply:
This example doesn't work (for unrelated reasons: `select` with a column after a row fails with `Select only allows row numbers for rows`), but I see your point.

We could fix this in a separate PR, but it would be a breaking change (similar to adding `?`, cf. #14233), so maybe @fdncred should chime in.

Reply:
I'm not following? Rows should be ints and Columns should be strings. Are we talking about changing that? If so, in what way?

Reply:
> Seems like `$confusing | get $cellp` works but `$confusing | select $cellp` doesn't.

That's because `select` doesn't let you mix rows and columns (see [this check](https://github.com/nushell/nushell/blob/main/crates/nu-command/src/filters/select.rs#L187)).

And I think this actually makes the point raised by @sholderbach not a problem, because the column names produced by `select` will _never_ contain row indices. e.g. if `select` produces a column `a.0.b`, that `0` came from a pre-existing column name, so there's no point quoting it?

Reply:
Just my opinion, but I like the way `get` works, as far as following cell paths, better than how select works right now. I like the flexibility of being able to do `get 0.a` or `get a.0` and getting the same result.